src/test/crypto_tests.cpp

```diff
@@ -524,6 +525,16 @@ BOOST_AUTO_TEST_CASE(chacha20_testvector)
                  "fab78c9");
 }
 
+BOOST_AUTO_TEST_CASE(hkdf_hmac_sha256_l32_tests)
+{
+    // rfc5869 has no test vectors for HMAC_SHA256 length 32, the one below was created with python HKDF and compared against openssl
```

Comment:
I think shorter-length HMAC_HKDF can be obtained by truncating the output of a longer-length one. So you could take all of the RFC's test vectors and truncate them to 32 bytes?

Reply:
Good idea. Followed @sipa's advice and changed to the 3 rfc5869 SHA256 test vectors (truncated OKM to 32 bytes).